S/spglib/build_tarballs.jl

```diff
@@ -42,6 +42,7 @@ dependencies = [
     # systems), and libgomp from `CompilerSupportLibraries_jll` everywhere else.
     Dependency("CompilerSupportLibraries_jll"; platforms=filter(!Sys.isbsd, platforms)),
     Dependency("LLVMOpenMP_jll"; platforms=filter(Sys.isbsd, platforms)),
+    Dependency("GoogleTest_jll"; platforms=filter(Sys.isbsd, platforms)),
```

Comment:
This one seems to only have the library, not the cmake files. What I've seen around here is that they add the latest google release to the sources and in this case use `-DFETCHCONTENT_SOURCE_DIR_GTEST=$(echo ${WORKSPACE}/srcdir/googletest*/)`.

Refference: https://github.com/JuliaPackaging/Yggdrasil/blob/fc054dec336202eb777986817f0404062b5fa60c/N/NNPACK/build_tarballs.jl#L29

Reply:
Thanks for your input! I tried both on my computer and the CI system, with the same error:

```
CMake Error at test/CMakeLists.txt:34 (gtest_add_tests):
Unknown CMake command "gtest_add_tests".
Call Stack (most recent call first):
test/unit/CMakeLists.txt:7 (configure_spglib_tests)
```

Reply:
Can you run `tree` on `${WORKSPACE}/srcdir`? Also I am curious of a few `echo` and `cat` of `CMAKE_TARGET_TOOLCHAIN`

Reply:
Hi, I do not have `tree` installed on this container, but it looks like this:

```shell
sandbox:${WORKSPACE}/srcdir # ls -la ${WORKSPACE}/srcdir
total 0
drwxr-xr-x    4 root     root           128 Oct  2 08:11 .
drwxr-xr-x    7 root     root           224 Oct  2 08:10 ..
drwxr-xr-x   18 root     root           576 Oct  2 08:11 googletest
drwxr-xr-x   29 root     root           928 Oct  2 08:10 spglib
sandbox:${WORKSPACE}/srcdir # ls googletest
BUILD.bazel          CONTRIBUTING.md      LICENSE              WORKSPACE            docs                 googletest
CMakeLists.txt       CONTRIBUTORS         README.md            ci                   googlemock           googletest_deps.bzl
sandbox:${WORKSPACE}/srcdir # ls spglib/
CITATION.cff       COPYING            README.md          doc                pyproject.toml     src
CMakeLists.txt     ChangeLog          cmake              example            python             test
CMakePresets.json  Contributing.md    database           fortran            ruby
```

The directories should reproduce the following code structures: https://github.com/google/googletest/tree/v1.14.0 and https://github.com/spglib/spglib/tree/v2.1.0

```shell
sandbox:${WORKSPACE}/srcdir # echo $CMAKE_TARGET_TOOLCHAIN
/opt/toolchains/x86_64-linux-gnu-libgfortran3-cxx03/target_x86_64-linux-gnu.cmake
sandbox:${WORKSPACE}/srcdir # cat $CMAKE_TARGET_TOOLCHAIN
# CMake toolchain file for gcc running on x86_64-linux-gnu-libgfortran3-cxx03
set(CMAKE_HOST_SYSTEM_NAME Linux)
set(CMAKE_HOST_SYSTEM_PROCESSOR x86_64)
execute_process(COMMAND /bin/uname -r OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
set(CMAKE_SYSTEM_NAME Linux)
set(CMAKE_SYSTEM_PROCESSOR x86_64)
execute_process(COMMAND /usr/bin/uname -r OUTPUT_VARIABLE CMAKE_SYSTEM_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)

set(CMAKE_SYSROOT /opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/)
set(CMAKE_INSTALL_PREFIX $ENV{prefix})

set(CMAKE_C_COMPILER   /opt/bin/x86_64-linux-gnu-libgfortran3-cxx03/x86_64-linux-gnu-gcc)
set(CMAKE_CXX_COMPILER /opt/bin/x86_64-linux-gnu-libgfortran3-cxx03/x86_64-linux-gnu-g++)
set(CMAKE_Fortran_COMPILER /opt/bin/x86_64-linux-gnu-libgfortran3-cxx03/x86_64-linux-gnu-gfortran)

set(CMAKE_LINKER  /opt/bin/x86_64-linux-gnu-libgfortran3-cxx03/x86_64-linux-gnu-ld)
set(CMAKE_OBJCOPY /opt/bin/x86_64-linux-gnu-libgfortran3-cxx03/x86_64-linux-gnu-objcopy)

set(CMAKE_AR     /opt/bin/x86_64-linux-gnu-libgfortran3-cxx03/x86_64-linux-gnu-ar)
set(CMAKE_NM     /opt/bin/x86_64-linux-gnu-libgfortran3-cxx03/x86_64-linux-gnu-nm)
set(CMAKE_RANLIB /opt/bin/x86_64-linux-gnu-libgfortran3-cxx03/x86_64-linux-gnu-ranlib)

if( $ENV{CC} MATCHES ccache )
    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
endif()
```

Reply:
Ok, please change `-DCMAKE_TARGET_TOOLCHAIN` to `-DCMAKE_TOOLCHAIN_FILE`, otherwise it has no effect.

I am baffled by why this issue is not present in the upstream test. Apparently there is a missing `include(GoogleTest)` which seems to be bundled by cmake and not `googletest`. But the tests compiled fine both with `FetchContent` (the CI runners) and with `find_package` (the Fedora CIs). I am not sure where there is a hidden `include(GoogleTest)` that we are implying :worried:. What cmake version is used there?

Reply:
```shell
sandbox:${WORKSPACE}/srcdir/spglib # cmake --version
cmake version 3.21.3
```

Reply:
Can you try to apply this patch?
```patch
From a597c9120c168e34ed4452d14fc8d4b59e2f4cd4 Mon Sep 17 00:00:00 2001
From: Cristian Le <cristian.le@mpsd.mpg.de>
Date: Mon, 2 Oct 2023 11:16:33 +0200
Subject: [PATCH] Add missing `include`

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>
---
 test/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 19d565e..14517fa 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,4 +1,5 @@
 enable_language(CXX)
+include(GoogleTest)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
-- 
2.41.0
```

A more step-by-step:
- Add a file, e.g. `0001-Add-missing-include.patch` with that content.
- `patch -p1 < /path/to/0001-Add-missing-include.patch` from the `${WORKSPACE}/srcdir/spglib` directory. (I think `-p1` is right in this context)

Reply:
I tried using this patch but errors still occur, thus I tried to build GoogleTest first, but still was unsuccessful. See https://github.com/JuliaPackaging/Yggdrasil/pull/7310#issuecomment-1743863251